### PR TITLE
Reland [clang][AArch64] Add getHostCPUFeatures to query for enabled f…

### PR DESCRIPTION
**Pull request by ElvinaYakubova**
…eatures in cpu info

Relands #97749. Fixed test by adding additional checks for system linux and target == host.

### CODE REVIEW

File: clang/test/lit.cfg.py [hunk 1]
```diff
@@ -210,6 +210,9 @@ def have_host_clang_repl_cuda():
 config.substitutions.append(("%host_cc", config.host_cc))
 config.substitutions.append(("%host_cxx", config.host_cxx))
 
+# Determine whether the test target is compatible with execution on the host.
+if config.host_arch in config.target_triple:
+    config.available_features.add("host-arch-compatible")
```
Comment (sjoerdmeijer) [inline clang/test/lit.cfg.py:215]: One nit that could be considered before committing: maybe `host-target-compatible` is slightly more descriptive?

**Review by sjoerdmeijer — APPROVED** [review]
Looks like a good fix to me. 
Please wait a day in case @davemgreen has comments, and it is better to commit in the morning anyway. :)

For convenience, the diff with the previous version is this first line in the test:

 ```
// REQUIRES: aarch64-registered-target,system-linux,host-arch-compatible
```

And this in the lit config file:

```
if config.host_arch in config.target_triple:
    config.available_features.add("host-arch-compatible")
```

**Comment by davemgreen** [comment]
I'm not sure I understood the original error properly but if we can come up with a fix like this it sounds sensible to me. Is the test still failing in the pre-commit tests? You might need something stronger - like only running the test on aarch64 host machines. There are some other tests for -mcpu=native that have comments that it is difficult to test.

File: clang/test/lit.cfg.py (hunk 1 above)
Comment (ElvinaYakubova) [reply clang/test/lit.cfg.py]: Thanks for the suggestion, renamed it

**Comment by ElvinaYakubova** [comment]
@davemgreen all tests are shown as passed, and I noticed the same failure on some other PRs. Can this be an infrastructure problem then? 
Regarding the original error - it was showing "unsupported argument 'native' to option '-mcpu='" for the cases where the target was AArch64 and the host was X86, that's why I added a check to compare host and target. And also there was a crash for Darwin systems, since we added this feature only for AArch64+linux case.

**Comment by davemgreen** [comment]
You have to go searching in the test output for the aarch64-mcpu-native.c test. The bots run clang tests separately to llvm, with a lot of verbose output, which means the test that is failing might not be at the end of the output. It looks like it still has the same error message in the latest test run.

I think I see what is going on - you might have to make sure that the machine this runs on is AArch64 (and maybe linux), as getHostCPUName is ifdef'd per the host architecture. It will be difficult to test elsewhere.

**Comment by ElvinaYakubova** [comment]
Oh yes, you were right, thanks for the explanation. 
I modified the test to run it only in case the host is AArch64.

**Review by davemgreen — APPROVED** [review]
Thanks that does look like it did it.

**Pull request merged**